include/circt/Dialect/OM/OMOps.td

```diff
@@ -140,4 +140,22 @@ def ConstantOp : OMOp<"constant",
   let hasFolder = true;
 }
 
+def ListCreateOp : OMOp<"list_create", [Pure, SameTypeOperands]> {
+  let summary = "Create a list of values";
+  let description = [{
+    Creates a list from a sequence of inputs.
+
+    ```
+    %list = om.list_create %a, %b, %c : !om.ref
+    ```
+  }];
+
+  let arguments = (ins Variadic<AnyType>:$inputs);
+  let results = (outs
+    ListType:$result
+  );
+
+  let hasCustomAssemblyFormat = 1;
```

Comment:
Would it be possible to use the result type in the textual form and use the assembly format in ODS? Eh on second thought this is how `hw.array_create` works, so this seems fine. It would still be nice to use the assembly format string in ODS, if it's possible to extract the type of the first operand in ODS.

Reply:
I think we should keep this in sync with `hw.array_create`. I don't know how to get ODS to specify the element type and check. I am pretty sure we can build a constraint to validate that arguments match the element type of an array, but then we'd have to specify the full list type for the list.